### ISSUE TITLE
yarn/npm install fails on Win 10 (running on Parallels VM on macOS)

### ISSUE THREAD

**Issue by phoenisx** [getsentry/sentry-cli#1308]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which SDK and version?
- Node: `12.22.12`
- Npm: `6.14.16`
- Yarn: `1.22.19`
- @sentry/cli: `1.74.4`

### Steps to Reproduce

Error very similar to https://github.com/getsentry/sentry-cli/issues/1246

Running `yarn install` fails, same with `npm install`

### Steps
`yarn install` with ENV variable `SENTRYCLI_USE_LOCAL` set to `"1"`
  - Compiled the project to generate local binary and made the command `sentry-cli.exe` global.
  - Running `node .\node_modules\@sentry\cli\scripts\install.js` works,
    but running with `yarn install` above script throws error.

For this step, I debugged the issue further and found that the package [which](https://github.com/npm/node-which) used by the install script returns a different `binPath` when called with `yarn`/`npm` and `node` directly.

With `yarn`/`npm`: `which.sync('sentry-cli')` returns wrong path -> `C:\{path\to\project}\node_modules\.bin\sentry-cli.CMD`.
while, with `node`: `which.sync('sentry-cli')` returns -> `C:\{path\to}\sentry-cli.exe`

**Error trace from `yarn install --verbose`** 👇🏽 
```sh
[4/19] ⠁ @sentry/cli
verbose 1686.3041369 Error: C:\{path\to\project}\node_modules\@sentry\cli: Command failed.
Exit code: 1
Command: node ./scripts/install.js
Arguments:
Directory: C:\{path\to\project}\node_modules\@sentry\cli
Output:
info sentry-cli Using local binary: C:\{path\to\project}\node_modules\.bin\sentry-cli.CMD
Error: spawn UNKNOWN
    at ProcessTermError.ExtendableBuiltin (C:\nvm4w\v12.22.12\node_modules\yarn\lib\cli.js:721:66)
    at ProcessTermError.MessageError (C:\nvm4w\v12.22.12\node_modules\yarn\lib\cli.js:750:123)
    at new ProcessTermError (C:\nvm4w\v12.22.12\node_modules\yarn\lib\cli.js:790:113)
    at ChildProcess.<anonymous> (C:\nvm4w\v12.22.12\node_modules\yarn\lib\cli.js:25787:17)
    at ChildProcess.emit (events.js:314:20)
    at maybeClose (internal/child_process.js:1022:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
error C:\{path\to\project}\node_modules\@sentry\cli: Command failed.
Exit code: 1
Command: node ./scripts/install.js
Arguments:
Directory: C:\{path\to\project}\node_modules\@sentry\cli
```

**Comment by kamilogorek** [comment]
At the first glance, it looks like the issue with `nvm` (https://github.com/coreybutler/nvm-windows), which is overriding yarn/npm env variables, and changes where global binaries are picked from. See note from their repo:

> Please note that any global npm modules you may have installed are not shared between the various versions of node.js you have installed. Additionally, some npm modules may not be supported in the version of node you're using, so be aware of your environment as you work.

**Comment by phoenisx** [comment]
I chose to build the `sentry-cli` binary and use it on Parallels because running `yarn install` (with `nvm`) failed initially (irrespective of the `SENTRYCLI_USE_LOCAL` ENV value).

Debugged a lot, but was not able to fix the error when `SENTRYCLI_USE_LOCAL` was set to `"1"`.
Using official NodeJS (and not `nvm`) worked for me when `SENTRYCLI_USE_LOCAL` was set to `"0"`.

I found two things in my testing:

- `yarn install` works properly without any issue for `@sentry/cli`
  (with `SENTRYCLI_USE_LOCAL` set to `"0"`) when using official NodeJS
  installer and not `nvm`.
- `yarn install` fails when `SENTRYCLI_USE_LOCAL` is set to `"1"`, no matter
  which NodeJS I used, Official or using `nvm` both caused the same issue.

I found that the behaviour that `which` provides can be easily achieved by checking the OS platform manually:

- UNIX like platforms: spawn(`which`)
- DOS-like platforms: spawn(`where.exe`)

Anyways, Thanks for your time and a quick response 🙇🏽. I was able to unblock myself 🎉
